print, repr ehancements

stepancheg opened the pull request:
- rewrite to use test framework
- fix printing of static values
- fix printing of various strings
- added CStringRef from string literal constructor
- fix print null CStringRef
- fix repr null CStringRef

jckarter commented on lib-clay/printer/printer.clay:
```diff
@@ -366,7 +366,8 @@ overload printReprTo(stream, x:E) {
 overload printReprTo(stream, x:Static[X]) {
-    printTo(stream, StaticName(x));
+    printTo(stream, "#");
+    printReprTo(stream, unwrapStatic(x));
```
Do symbols and string literals still print without `#`?

stepancheg replied:
Fixed that.

stepancheg committed:
repr: fix printing of static values

stepancheg committed:
repr: rewrite tests to use test framework

stepancheg committed:
repr: correct repr for various strings

stepancheg committed:
CStringRef constructor from string literal

stepancheg commented:
Added correct printing of various strings.

stepancheg committed:
printer: print null CStringRef

Previously code crashed with segmentation fault.

"(null)" is chosen for compatibility with Linux printf implementation.

stepancheg committed:
repr: print null CStringRef
